pkgs/applications/misc/joplin-desktop/default.nix

```diff
@@ -8,17 +8,11 @@ let
   inherit (stdenv.hostPlatform) system;
   throwSystem = throw "Unsupported system: ${system}";
 
-  suffix = {
-    x86_64-linux = "AppImage";
-    x86_64-darwin = "dmg";
-  }.${system} or throwSystem;
+  suffix = if stdenv.isDarwin then "dmg" else "AppImage";
```

Comment:
Please edit on top of the original one.

Reply:
@wegank `darwin` artifact is going to be the same for both architectures. Is it really necessary to duplicate it? I am open to better ways of doing this, but duplicating seems not the best one. What do you think?

Reply:
You can do something like that with `suffix = rec {`:

https://github.com/NixOS/nixpkgs/blob/6e9531b9131da0c6542119f8d700e76d9c2eb1f4/pkgs/tools/networking/ookla-speedtest/default.nix#L28

Reply:
I have re-write the changes to generate on `src` using the same strategy as in the example. It looks much cleaner, I think 🤔